cfpb/consumerfinance.gov
# Filterable Search Enhancements

Enhance filterable search by expanding the ngram tokenizer we use for matching terms, as well as introducing text based searching against the content, preview description, and topics of the results. 


---



## Additions

- New tokenizer that isn't shared with autocomplete - expands the size of which we can match substrings on search results.
- New searchable fields - content, preview description, and tag names.


## Changes

- Now uses a multimatch search to query across several fields instead of just the title field.


## How to test this PR

1. Reindex your elasticsearch cluster
2. Search a term not found in the title but found in the description (I found blogs on pandemic end up resulting in more results)
3. Confirm you see additional results based on new matching logic.


## Screenshots


## Notes and todos

-


## Checklist



  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance
